src/config/configure.in

```diff
@@ -1745,20 +1810,53 @@ then
 fi
 
 
-if [test "$hypre_using_cuda" = "yes" || test "$hypre_using_device_openmp" = "yes" || test "$hypre_using_um" = "yes"]
+if [ test "x$hypre_using_um" = "xyes"]
 then
-   AC_CHECK_HEADERS(["${CUDA_HOME}/include/cuda.h"], [hypre_found_cuda=yes; HYPRE_CUDA_PATH=${CUDA_HOME}])
+  if [test "$hypre_using_cuda" = "yes" || test "$hypre_using_device_openmp" = "yes"]
+  then
+     AC_CHECK_HEADERS(["${CUDA_HOME}/include/cuda.h"], [hypre_found_cuda=yes; HYPRE_CUDA_PATH=${CUDA_HOME}])
+
+     if test "x$hypre_found_cuda" != "xyes"
+     then
+        AC_CHECK_HEADERS(["${CUDA_PATH}/include/cuda.h"], [hypre_found_cuda=yes; HYPRE_CUDA_PATH=${CUDA_PATH}])
+     fi
+
+     if test "x$hypre_found_cuda" != "xyes"
+     then
+        AC_MSG_ERROR([unable to find cuda.h ... Ensure that CUDA_HOME or CUDA_PATH is set])
+     fi
+  dnl if not using cuda, make sure we're using hip
+  dnl if not error. if we are, HIP will be checked below.
+  else
+    if [test "x$hypre_using_hip" != "xyes"]
+    then
+       AC_MSG_ERROR([Asked for unified memory, but not using CUDA, HIP, or device OpenMP!])
+    fi
+  fi
+fi dnl hypre_using_um
+
+dnl *********************************************************************
+dnl * Check for HIP header
+dnl *********************************************************************
+
+dnl If the user has requested to use HIP, we first check the environment
+dnl for ROCM_PATH to point at the ROCm installation. If that is not found,
+dnl then we default to `/opt/rocm`.
+dnl
+dnl TODO: Add an ARG_WITH for rocm so the user can control the ROCm path
```

Comment:
@pbauman I like this idea of "Adding an ARG_WITH", which has been done in another PR for CUDA.